PR: https://github.com/odoo/odoo/pull/96280

From: 985e49bdb5e22fa197ba267aa41d7a8ba7e50550
From: Romain Derie
Date: 2022-07-26 13:48:43

Breaking data model changes scores: del:5 + add:0, change matches:
-    google_management_client_id = fields.Char(
-    google_management_client_secret = fields.Char(
-    has_google_analytics_dashboard = fields.Boolean(
-    google_management_client_id = fields.Char('Google Client ID')
-    google_management_client_secret = fields.Char('Google Client Secret')

Total Changes: 27

[FIX] website: remove Google Analytics dashboard (deprecated)

==== Short version ====

Google is deprecating Universal Analytics in July 2023 and Google
Sign-In in March 2023. Google Analytics Embed API is based on Sign-In,
meaning it won't work anymore. It actually already doesn't work anymore
for accounts created somewhere after mid-2020 apparently.
There is no plan for now for Google to allow Analytics 4 dashboard to be
embed in external website.
We therefore can't do anything to keep the Google Analytics dashboard in
Odoo.
In previous stable version, it was kept but is displaying a warning
about it (as until mid 2023 old accounts can still embed it).
All this is about the embed dashboard, not the tracking in itself for
which Odoo is already adapted in Odoo 15.0 for Analytics 4.

==== Detailed version (following short version, read it first)  ====

* Universal Analytics EOL July 2023, see [1].
* It will be replaced by Analytics 4 for which Odoo is already ready and
  actually using it since version 15.0 with [2].
* Google Sign-In EOL March 2023, see [3]. Analytics Embed API was based
  on it, it won't work anymore.
* There is no plan (for now) for Google to allow Analytics 4 to be able
  to be embed in external websites. They seem to just have dropped the
  "feature".
  This was confirmed by Google here [4] and indirectly here [5] in the
  DOC:
  `Note: This API does not support Google Analytics 4 (GA4) properties`
* While the EOL is planed for 2023, the dashboard integration is already
  not working anymore for new accounts.
* Old projects/keys/accounts can still embed their analytics dashboard.
  The threshold seems to be somewhere mid-2020, according to [6].
  It seems to be accurate as my own key from 2018 still works, while my
  keys from 2021 do not.

==== Fix ====

* In stable, warn user about it in their Odoo Analytics dashboard (this
  PR) and also add a warning about that on the doc.
* In master, simply drop the whole google analytics dashboard
  integration and remove the doc about it, see [7].

==== Useful links ====

[1]: https://support.google.com/analytics/answer/11583528?hl=en
[2]: https://github.com/odoo/odoo/commit/78bc86cbeccfc5df16218aee2b0d7c501e5c05b5
[3]: https://developers.googleblog.com/2022/03/gis-jsweb-authz-migration.html
[4]: https://issuetracker.google.com/issues/233738709?pli=1
[5]: https://developers.google.com/analytics/devguides/reporting/embed/v1
[6]: https://support.google.com/analytics/answer/11583832
[7]: https://www.odoo.com/documentation/15.0/applications/websites/website/optimize/google_analytics_dashboard.html

Finally, note that it means that from July 2023 to Octobre 2023, while
Odoo 14.0 is still supported, Google Analytics won't work anymore in
that version as it will still be designed for Universal Analytics and
not Analytics 4.

opw-2710910
opw-2855405
opw-2881515
opw-2892370
task-2790245
task-2820890

closes odoo/odoo#96280

X-original-commit: d065595f77790fb5ab9480f6de5b88549352324b
Related: odoo/enterprise#29666
Related: odoo/upgrade#3698
Related: odoo/documentation#2499
Signed-off-by: Romain Derie (rde) <rde@odoo.com>

================================= pseudo patch: =================================

--- a/addons/website/models/res_config_settings.py
+++ b/addons/website/models/res_config_settings.py
@@ -64,14 +64,6 @@ class ResConfigSettings(models.TransientModel):
         'Google Analytics Key',
         related='website_id.google_analytics_key',
         readonly=False)
-    google_management_client_id = fields.Char(
-        'Google Client ID',
-        related='website_id.google_management_client_id',
-        readonly=False)
-    google_management_client_secret = fields.Char(
-        'Google Client Secret',
-        related='website_id.google_management_client_secret',
-        readonly=False)
     google_search_console = fields.Char(
         'Google Search Console',
         related='website_id.google_search_console',
@@ -115,10 +107,6 @@ class ResConfigSettings(models.TransientModel):
         "Google Analytics",
         compute='_compute_has_google_analytics',
         inverse='_inverse_has_google_analytics')
-    has_google_analytics_dashboard = fields.Boolean(
-        "Google Analytics Dashboard",
-        compute='_compute_has_google_analytics_dashboard',
-        inverse='_inverse_has_google_analytics_dashboard')
     has_google_search_console = fields.Boolean(
         "Console Google Search",
         compute='_compute_has_google_search_console',
@@ -187,21 +175,8 @@ class ResConfigSettings(models.TransientModel):
         for config in self:
             if config.has_google_analytics:
                 continue
-            config.has_google_analytics_dashboard = False
             config.google_analytics_key = False
 
-    @api.depends('website_id')
-    def _compute_has_google_analytics_dashboard(self):
-        for config in self:
-            config.has_google_analytics_dashboard = bool(config.google_management_client_id)
-
-    def _inverse_has_google_analytics_dashboard(self):
-        for config in self:
-            if config.has_google_analytics_dashboard:
-                continue
-            config.google_management_client_id = False
-            config.google_management_client_secret = False
-
     @api.depends('website_id')
     def _compute_has_google_search_console(self):
         for config in self:

--- a/addons/website/models/website.py
+++ b/addons/website/models/website.py
@@ -108,8 +108,6 @@ class Website(models.Model):
     has_social_default_image = fields.Boolean(compute='_compute_has_social_default_image', store=True)
 
     google_analytics_key = fields.Char('Google Analytics Key')
-    google_management_client_id = fields.Char('Google Client ID')
-    google_management_client_secret = fields.Char('Google Client Secret')
     google_search_console = fields.Char(help='Google key, or Enable to access first reply')
 
     google_maps_api_key = fields.Char('Google Maps API Key')
